src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -169,8 +166,22 @@ private long internalCheck(EC2Computer computer) {
             }
 
             //on rare occasions, AWS may return fault instance which shows running in AWS console but can not be connected.
-            //need terminate such fault instance by {@link #STARTUP_TIMEOUT}
-            if (computer.isOffline() && uptime < TimeUnit.MINUTES.toMillis(STARTUP_TIMEOUT)) {
+            //need terminate such fault instance.
+            // An instance may also fail running user data scripts and
+            // need to be cleaned up.
+            if (computer.isOffline()){
+                EC2AbstractSlave node = computer.getNode();
+                if (Objects.isNull(node)){
```

Comment:
```suggestion
                if (node == null){
```

The `Objects::isNull` method is mostly useful as a function reference.